src/dimension_slice.c

```diff
@@ -638,23 +799,14 @@ ts_dimension_slice_delete_by_dimension_id(int32 dimension_id, bool delete_constr
 int
 ts_dimension_slice_delete_by_id(int32 dimension_slice_id, bool delete_constraints)
 {
-	ScanKeyData scankey[1];
+	FormData_dimension_slice form;
```

Comment:
don't we also have to modify ts_dimension_slice_delete_by_dimension_id function (line 777) to follow the same pattern?

Reply:
I've added tuple locking to ts_dimension_slice_delete_by_dimension_id and check for lock result so effectively we get the same behavior as with the custom workflow. I don't want to re-implement the scanner for deletes since they are not at issue here.

Refactor was mostly directed at updates due to concurrent operations creating phantom updates.